pkg/controllers/state/suite_test.go

```diff
@@ -1241,6 +1245,9 @@ var _ = Describe("Cluster State Sync", func() {
 			ExpectReconcileSucceeded(ctx, nodeClaimController, client.ObjectKeyFromObject(nodeClaim))
 		}
 		Expect(cluster.Synced(ctx)).To(BeFalse())
+		metric, found := FindMetricWithLabelValues("karpenter_cluster_state_unsynced_time_seconds", map[string]string{})
```

Comment:
I think this would work because we also reset this metric after each test. So even if it succeeded in previous iteration, the values would get cleaned up.

Reply:
Updated it to use fake clock